lib/reek/context/code_context.rb

```diff
@@ -136,24 +132,30 @@ def matches?(candidates)
       end
 
       def full_name
-        exp.full_name(context ? context.full_name : '')
+        exp.full_name(parent ? parent.full_name : '')
       end
 
       def config_for(detector_class)
-        context_config_for(detector_class).merge(
+        parent_config_for(detector_class).merge(
           configuration_via_code_commment[detector_class.smell_type] || {})
       end
 
-      def track_visibility(visibility, names)
-        visibility_tracker.track_visibility children: children,
-                                            visibility: visibility,
-                                            names: names
-      end
-
       def number_of_statements
         statement_counter.value
       end
 
+      def singleton_method?
+        false
+      end
+
+      def instance_method?
+        false
+      end
+
+      def apply_current_visibility(_current_visibility)
+        # Nothing to do by default
```

Comment:
I tried this just now and it doesn't work, since the logic for setting the visibility is different depending on whether the current visibility is used or the visibility is set explicitly using e.g. `private` with arguments.